software/firmware/europi_script.py

```diff
@@ -52,23 +54,92 @@ def display_name(cls):
         """
         return cls.__qualname__
 
-    #
-    # private functions for now, with the intention that they are eventually added to the public API
-    #
+    # Save State Methods
+
+    @property
+    def _state_filename(self):
+        return f"saved_state_{self.__class__.__qualname__}.txt"
+
+    def save_state(self, state: str):
```

Comment:
I'm wondering if this should be `save_state_str()` to avoid the override mentioned in my comment in `menu.md`